src/detectors/ScanningStrategyDetector.ts

```diff
@@ -67,21 +67,21 @@ export class ScanningStrategyDetector implements IDetector<string, ScanningStrat
       return undefined;
     }
 
-    if (true || remoteService.serviceType === ServiceType.github) {
+    if (remoteService.serviceType === ServiceType.github) {
       const parsedUrl = gitUrlParse(remoteService.remoteUrl);
 
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       let response: any;
       try {
         response = await this.gitHubClient.get(parsedUrl.owner, parsedUrl.name);
       } catch (error) {
-        if (error.status && error.status === 404) {
-          return AccessType.private;
-        }
-        throw error;
+        throw ErrorFactory.newInternalError('You passed bad credentials or non existing repo.');
```

Comment:
It's not an internal error, it's a new type of error. It could be ArgumentError. The error is because of the user's input, so it should be something like `Unauthorized` or `InvalidArguments`.

Reply:
We should catch 404 and 403 (or 401) errors and if the status code is different, then throw default error, because it isn't error of the user's input.

Reply:
Ah, I see. Tell me if I'm wrong but. I'll just change the  `throw ErrorFactory.newInternalError()` for  `throw ErrorFactory.newArgumentError()` ? Or I can also add new error to ErrorFactory, if you want to.

Reply:
I guess `argumentError` is enough.